src/js/stories/form-experiment/TextInputClassic.jsx

```diff
@@ -0,0 +1,108 @@
+import React, { Component, PropTypes } from 'react';
+
+
+export default class TextInputClassicStory extends Component {
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            defaultValue: 'default value passed before mounting',
+        };
+
+        this.textRef = null;
+    }
+
+    _setDefaultValue(value) {
+        console.log('_setDefaultValue:', value);
+        // this doesn't work since defaultValue is only read from when TextInputClassic mounts
+        this.setState({
+            defaultValue: value,
+        });
+    }
+
+    _setValue(value) {
+        console.log('_setValue:', value);
+        this.textRef.setValue('setValue: ' + value);
+    }
+
+    render() {
+        return (
+            <div style={{ display: 'flex', flexDirection: 'column', alignItems: 'flex-start', height: 300, justifyContent: 'space-between', padding: 10 }}>
+                <h1>Classic</h1>
+
+                <span>
+                    <TextInputClassic />
+                    &nbsp; updates as you type without extra code, without an initial value set
+                </span>
+
+                <span>
+                    <TextInputClassic defaultValue={this.state.defaultValue} />
+                    &nbsp; updates as you type without extra code, with an initial value set
+                </span>
+
+                <br />
+
+                <span style={{ background: '#FF9999'}}>
+                    <button onClick={() => this._setDefaultValue(new Date().toString())}>Set defaultValue</button>
+                    &nbsp; Won't work, since defaultValue is only read by text control when mounting
+                </span>
+
+                <span>
+                    <button onClick={() => this._setValue(new Date().toString())}>Call setValue</button>
+                    &nbsp; Will work, but requires a ref, and is less "reactive"
+                </span>
+
+                <TextInputClassic ref={ref => this.textRef = ref} defaultValue={this.state.defaultValue} />
+            </div>
+        );
+    }
+}
+
+
+class TextInputClassic extends Component {
+
+    propTypes: {
+        defaultValue: PropTypes.string,
+    };
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            value: '',
+        };
+    }
+
+    componentWillMount() {
+        this._initialize(this.props);
+    }
```

Comment:
@cliffmeyers 
> We could try to fix the bug by adding componentWillReceiveProps and letting it apply the defaultValue prop, but it won't work because any prop change would trigger this

well you could add an evaluation like `nextProps.deauftValue !== this.state.defaultValue` to `componentWillReceiveProps` or maybe even `shouldComponentUpdate`